modules/update-cvo-conditions.adoc

```diff
@@ -0,0 +1,65 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding-update-service.adoc
+
+[id="update-cvo-conditions_{context}"]
+= Understanding CVO conditions
+
+The `ClusterVersion` object sets the conditions that describe the state of the cluster-version operator (CVO).
+
+Failing:: When `Failing` is set to True, the CVO fails to reconcile the cluster with the desired release image. In all cases, the impact on the cluster is that the dependent nodes in the manifest graph may not be reconciled. The graph may be flattened, in which case there are no dependent nodes.
++
+Most reconciliation errors result in `Failing=True`, even though `ClusterOperatorNotAvaliable` has special handling.
+
+NoDesiredImage:: The CVO has not been given a release image to reconcile. When NoDesiredImage is True, it is a CVO coding error. The `desiredUpdate` field returns you to the current CVO's release image.
+
+ClusterOperatorNotAvailable:: `ClusterOperatorNotAvailable` is set when the CVO fails to retrieve the ClusterOperator from the cluster or when the retrieved ClusterOperator does not satisfy the reconciliation conditions.
++
+Unlike most manifest-reconciliation failures, this error does not immediately result in `Failing=True`. Under some conditions during installs and updates, the CVO will treat this condition as a `Progressing=True` condition and give the operator up to forty minutes to level before reporting `Failing=True`.
+
+RetrievedUpdates:: When `RetrievedUpdates` is set to True, the CVO successfully retrieves updates. When the `RetrievedUpdates` is set to False, the cluster is not able to retrieve recommended updates. When a CVO is unable to retrieve recommended updates the `CannotRetrieveUpdates` alert displays the reason for not retrieving recommended updates.
+
+// When `RetrievedUpdates` is False, reason will be set to explain why, as discussed in the following subsections.
+// In all cases, the impact is that the cluster will not be able to retrieve recommended updates, so cluster admins will need to monitor for available updates on their own or risk falling behind on security or other bugfixes.
+// When CVO is unable to retrieve recommended updates the CannotRetrieveUpdates alert will fire containing the reason.
+// This alert will not fire when the reason updates cannot be retrieved is NoChannel.
+
+NoUpstream:: No upstream server has been set to retrieve updates.
++
+Set `spec.upstream` in `ClusterVersion` to point to a link:https://github.com/openshift/cincinnati/blob/master/docs/design/openshift.md[Cincinnati server] to receive updates.
```

Comment:
We've had [a default][1] forever.  This should probably include the "This is a result of a CVO coding error." caveat.  But yeah, if we did have a bug in this space, explicitly setting `spec.upstream` might avoid the issue.

[1]: https://github.com/openshift/cluster-version-operator/blob/256eea42a0b99617f93902ac4e3f13a37a4b9380/pkg/cvo/availableupdates.go#L34